### PR TITLE
subgraph integration for ngraph_bridge

### PR DESCRIPTION
**Pull request by ashokei**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/ngraph/ngraph_compiler.cc [hunk 1]
```diff
@@ -228,7 +228,7 @@ void Compiler::CreateSubgraphNNVMNodes() {
       }
 
       // register compiled subgraph with nnvm
-      register_subgraph(sg);
+      // register_subgraph(sg);
```
Comment (mbrookhart) [inline src/ngraph/ngraph_compiler.cc]: remove dead code

File: src/ngraph/ngraph_subgraph.cc [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+#include "../operator/subgraph/default_subgraph_op.h"
+#include "ngraph_nnvm_ops.h"
+
+namespace ngraph_bridge {
+using namespace nnvm;
+using namespace mxnet;
+using namespace mxnet::op;
+
+std::shared_ptr<ngraph_bridge::Graph> get_ngraph(const NodeAttrs& attrs) {
+  auto& g = nnvm::get<ngraph_bridge::NGraphParam>(attrs.parsed).g;
```
Comment (mbrookhart) [inline src/ngraph/ngraph_subgraph.cc]: why assign it to a local reference instead of just returning directly?

File: src/ngraph/ngraph_subgraph.cc [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  compute_backward(ctx, ngraph_, inputs, req, outputs);
+}
+
+OpStatePtr CreateNgraphSubgraphOpState(const NodeAttrs& attrs, Context ctx,
+                                       const std::vector<TShape>& in_shapes,
+                                       const std::vector<int>& in_types) {
+  return OpStatePtr::Create<NgraphSubgraphOperator>(get_ngraph(attrs));
+}
+
+OpStatePtr CreateNgraphBackwardOpState(const NodeAttrs& attrs, Context ctx,
```
Comment (mbrookhart) [inline src/ngraph/ngraph_subgraph.cc]: Do we need separate forward and backward op state functions?

Comment (ashokei) [reply src/ngraph/ngraph_subgraph.cc]: i think backopstate creation may need to be separate when we use subgraph fully (as the node attrs of forward would not have ngraph yet, but backward might). currently both reuse same ngraph attrs.parsed.

File: src/ngraph/ngraph_compiler.cc (hunk 1 above)
Comment (ashokei) [reply src/ngraph/ngraph_compiler.cc]: will do

File: src/ngraph/ngraph_emitter.cc [hunk 4]
```diff
@@ -124,6 +124,7 @@ NgraphNodePtr Emitter::ReduceAxes(
 void Emitter::CreateUnaryOps() {
   ngraph_op_funcs_["Activation"] = [this](const NodePtr node) {
     auto act_type = node->orig_node_->attrs.dict["act_type"];
+    std::cout << act_type << std::endl;
```
Comment (ashokei) [inline src/ngraph/ngraph_emitter.cc]: left over ?
